src/core/ddsc/include/dds/ddsc/dds_psmx.h

```diff
@@ -32,365 +163,504 @@ extern "C" {
+ * A PSMX Instance in the "PSMX Instances" QoS may accept or reject a given DDS
+ * Topic/Reader/Writer based on information on the type and the QoS settings.  PSMX
+ * Instances not in the "PSMX Instances" QoS setting are not asked.
  *
- * @param[in] psmx_instance The PSMX instance.
- * @param[in] forwhat whether for a topic/writer/reader (UNSET if topic)
- * @param[in] data_type_props Data type properties
- * @param[in] qos  The QoS.
- * @returns true if the QoS is supported, false otherwise
+ * @param[in] psmx_instance    The PSMX Instance
+ * @param[in] forwhat          Whether for a topic/writer/reader (UNSET if topic)
+ * @param[in] data_type_props  Data type properties (see there)
+ * @param[in] qos              QoS of the DDS topic
+ * @returns True if the PSMX Instance is willing to handle this DDS entity, false otherwise
  */
 typedef bool (*dds_psmx_type_qos_supported_fn) (struct dds_psmx *psmx_instance, dds_psmx_endpoint_type_t forwhat, dds_data_type_properties_t data_type_props, const struct dds_qos *qos);
 
 /**
- * @brief Definition for function to create a topic
+ * @brief Definition of the function for creating a PSMX Topic (interface version 1)
+ * @ingroup psmx
+ *
+ * Definition for a function that is called to construct a new PSMX Topic in a PSMX Instance
+ * representing a new topic in the DDS Domain Entity.
+ *
+ * The PSMX Plugin is expected to represent a PSMX Topic using an extended version of the
+ * `dds_psmx_topic` structure.  It is required to initialize the generic fields using
+ * `dds_psmx_topic_init`.
+ *
+ * If `type_definition` is not a null pointer, it points into the Cyclone type library. A
+ * (default, C) serializer can be constructed using `ddsi_topic_descriptor_from_type`,
+ * an XTypes TypeObject using `ddsi_type_get_typeobj`.
+ *
+ * If the function returns failure, creation of the DDS Entity fails.
+ *
+ * @param[in] psmx_instance    The PSMX instance.
+ * @param[in] topic_name       The name of the topic to create
+ * @param[in] type_name        The name of the DDS data type for this topic
+ * @param[in] data_type_props  The data type properties for the topic's data type.
+ * @param[in] type_definition  Optional pointer to type definition in type library.
+ * @param[in] sizeof_type      In-memory size of a single instance of the type, 0 if unknown.
+ * @returns A new PSMX Topic structure, or NULL on error
+ */
+typedef struct dds_psmx_topic * (*dds_psmx_create_topic_type_fn) (
+    struct dds_psmx *psmx_instance,
+    const char *topic_name,
+    const char *type_name,
+    dds_data_type_properties_t data_type_props,
+    const struct ddsi_type *type_definition,
+    uint32_t sizeof_type);
+
+/**
+ * @brief Definition of the function for destructing and freeing a PSMX Topic
+ * @ingroup psmx
+ *
+ * Definition for a function that is called on deleting the topic in the DDS Domain.
+ * Called exactly once for each successful invocation of `dds_psmx_topic_new`, all
```

Comment:
`dds_psmx_topic_new` -> `create_topic` (or `create_topic_type`?)

And below: `dds_psmx_cleanup_topic_generic` -> `dds_psmx_topic_cleanup_generic`

(same in `dds__psmx.h` for `_int` variant)